lib/RoadizCoreBundle/src/Bag/DecoratedNodeTypes.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+declare(strict_types=1);
+
+namespace RZ\Roadiz\CoreBundle\Bag;
+
+use Doctrine\DBAL\Driver\PDO\PDOException;
+use http\Exception\RuntimeException;
+use RZ\Roadiz\Bag\LazyParameterBag;
+use RZ\Roadiz\Contracts\NodeType\NodeTypeResolverInterface;
+use RZ\Roadiz\CoreBundle\Entity\NodeType;
+use RZ\Roadiz\CoreBundle\Repository\NodeTypeDecoratorRepository;
+
+/**
+ * @method NodeType|null get(string $key, $default = null)
+ */
+final class DecoratedNodeTypes extends LazyParameterBag implements NodeTypeResolverInterface
+{
+    public function __construct(
+        private readonly NodeTypes $nodeTypesBag,
+        private readonly NodeTypeDecoratorRepository $repository,
+    ) {
+        parent::__construct();
+    }
+
+    protected function populateParameters(): void
+    {
+        $nodeTypes = $this->nodeTypesBag->all();
+        foreach ($nodeTypes as $nodeType) {
+            $decoratedNodeType = clone $nodeType;
+            try {
+                $nodeTypeDecorators = $this->repository->findByNodeType($decoratedNodeType);
+                foreach ($nodeTypeDecorators as $nodeTypeDecorator) {
+                    $nodeTypeDecorator->applyOn($decoratedNodeType);
+                }
+                $this->parameters[$decoratedNodeType->getName()] = $decoratedNodeType;
+                $this->parameters[$decoratedNodeType->getSourceEntityFullQualifiedClassName()] = $decoratedNodeType;
+            } catch (PDOException $e) {
+                throw new RuntimeException($e->getMessage(), $e->getCode(), $e);
+            }
```

Comment:
If you catch PDOException, you should do nothing instead of throwing back a new exception.

This is failing your tests!


You should only set the cloned nodetype into parameters array when DB fails.